### PR TITLE
[1.12] Expose EntityRenderer.thirdPersonDistance

### PR DESCRIPTION
**Pull request by alexander0153**
Use-case:
Mods that add large ridable entities can increase the third-person camera distance. This could keep the camera from clipping into the entity.

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch [hunk 1]
```diff
@@ -1,5 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/client/renderer/EntityRenderer.java
 +++ ../src-work/minecraft/net/minecraft/client/renderer/EntityRenderer.java
+@@ -94,7 +94,7 @@
+     private Entity field_78528_u;
+     private final MouseFilter field_78527_v = new MouseFilter();
+     private final MouseFilter field_78526_w = new MouseFilter();
+-    private final float field_78490_B = 4.0F;
++    public float field_78490_B = 4.0F;
```
Comment (PhiProven) [inline patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch]: I think you should use `forge_at.cfg` for this.

**Review by PhiProven — CHANGES_REQUESTED** [review]

**Comment by alexander0153** [comment]
I moved the access modifier changes into the `forge_at.cfg` file.

**Comment by mezz** [comment]
Looks good to me.
Please create a small test mod which sets the `thirdPersonDistance` to show that it works as expected.
You can find the other test mods in `net.minecraftforge.debug`.
Your test mod should be disabled by default, see `AdvancementEventTest` as an example.

**Comment by alexander0153** [comment]
Cool! I also added a test mod now.

**Comment by Gegy** [comment]
A common problem I've noticed with changing third person distance is mods can override each other easily. Would there be any way this could be more friendly to this kind of situation? Maybe a multiplier system instead of directly setting it, or possibly priority for which distance should be selected?
You could probably assume that the largest distance would be desired.

**Comment by alexander0153** [comment]
@gegy1000 I was originally thinking of an event, similar to `FOVUpdateEvent`, with which you could set a distance multiplier. However, I am not sure about performance implications of firing an event every frame, especially for a feature that only a few mods would use.

**Comment by Gegy** [comment]
@alexander0153 Understandable. I don't think it needs to be fired every frame though -- it could be done each tick (interpolation is already done).

**Comment by alexander0153** [comment]
I added a new event: `CameraDistanceUpdateEvent`.

If a mod wants to increase the distance between camera and entity, it can call the `multiplyDistance(float multiplier)` method of that event (see the test mod).

As @gegy1000 suggested, this would allow multiple mods to change the distance without overwriting each other's changes.

**Comment by mezz** [comment]
Having this event fired every tick seems a bit much compared to the old version.

I think the case of two mods competing on the camera distance needs to be considered more carefully.
If one mod wants to zoom out 2x and another mod wants the same thing, you end up with 4x and nobody really gets what they wanted.
If one mod wants 4x and another just always sets it to 2x then it will depend on whoever went last.

We should try to find a good way for mods to work together on this event, and if that doesn't work I think we should prefer just setting the field.

**Comment by williewillus** [comment]
Anytime you have stacking modifiers on a single numeric value, attributes are a great system to use. Although, in this case the number is client side only which complicates things a bit.

**Comment by alexander0153** [comment]
I have reverted my changes for the new event. However, I don't know how to improve on this. I have not worked with attributes yet, but from what I see they are mainly used for entities. This is my first contribution to Minecraft Forge, so I am not sure what the best way to implement this is.

To me, setting the field directly seems like the simplest solution, especially if only very few mods would use this feature.

**Pull request closed without merging**